examples/01-sage-game/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Program } from "@project-serum/anchor";
+import { Connection, Keypair } from "@solana/web3.js";
+
+import { readAllFromRPC } from "@staratlas/data-source";
+import { Game, SAGE_IDL } from "@staratlas/sage";
+
+import { newConnection, newAnchorProvider } from "../../shared/anchor-setup";
+import { loadKeypair } from "../../shared/wallet-setup";
+
+const SAGE_PROGRAM_ID = "SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE"
+
+const mainFunction = async (connection: Connection, myWallet: Keypair) => {
+    console.log("Example 01: Sage Game");
+
+    const provider = newAnchorProvider(connection, myWallet);
+    const sageProgram = new Program(
+        SAGE_IDL,
+        SAGE_PROGRAM_ID,
+        provider,
+    );
+
+    const games = await readAllFromRPC(
+        connection,
+        sageProgram as any,
+        Game,
+        'processed',
+        [],
+    );
+
+    if (games.length === 0) {
+        throw 'no game found';
```

Comment:
It's better to `throw Error('no game found')`

catching exceptions is not type-safe, plus you loose information (e.g. the stack) if you throw strings

Reply:
See commit [7100121](https://github.com/staratlasmeta/star-atlas-cookbook/pull/1/commits/710012175c04d238db757ca99467af89103325c0)